src/Utilities/Blaze.hpp

```diff
@@ -41,4 +40,76 @@ constexpr bool useDefaultInitialization = false;
 
 // Disable HPX parallelization
 #define BLAZE_HPX_PARALLEL_MODE 0
+
+// Disable all padding
+#define BLAZE_USE_PADDING 0
+
+#define BLAZE_USE_STREAMING 1
+#define BLAZE_USE_OPTIMIZED_KERNELS 1
+#define BLAZE_USE_DEFAULT_INITIALIZATON 0
 /// \endcond
+
+namespace blaze {
+template <typename T>
+BLAZE_ALWAYS_INLINE SIMDdouble step_function(const SIMDf64<T>& v) noexcept
+#if BLAZE_AVX512F_MODE || BLAZE_MIC_MODE
+{
+  return _mm512_set_pd((*v).eval().value[7] < 0.0 ? 0.0 : 1.0,
+                       (*v).eval().value[6] < 0.0 ? 0.0 : 1.0,
+                       (*v).eval().value[5] < 0.0 ? 0.0 : 1.0,
+                       (*v).eval().value[4] < 0.0 ? 0.0 : 1.0,
+                       (*v).eval().value[3] < 0.0 ? 0.0 : 1.0,
+                       (*v).eval().value[2] < 0.0 ? 0.0 : 1.0,
+                       (*v).eval().value[1] < 0.0 ? 0.0 : 1.0,
+                       (*v).eval().value[0] < 0.0 ? 0.0 : 1.0);
+}
+#elif BLAZE_AVX_MODE
+{
+  return _mm256_set_pd((*v).eval().value[3] < 0.0 ? 0.0 : 1.0,
+                       (*v).eval().value[2] < 0.0 ? 0.0 : 1.0,
+                       (*v).eval().value[1] < 0.0 ? 0.0 : 1.0,
+                       (*v).eval().value[0] < 0.0 ? 0.0 : 1.0);
+}
+#elif BLAZE_SSE2_MODE
+{
+  return _mm_set_pd((*v).eval().value[1] < 0.0 ? 0.0 : 1.0,
+                    (*v).eval().value[0] < 0.0 ? 0.0 : 1.0);
+}
+#else
+{
+  return SIMDdouble{(*v).value < 0.0 ? 0.0 : 1.0};
+}
+#endif
+
+BLAZE_ALWAYS_INLINE double step_function(const double v) noexcept {
+  return v < 0.0 ? 0.0 : 1.0;
+}
+
+struct StepFunction {
+  explicit inline StepFunction() = default;
+
+  template <typename T>
+  BLAZE_ALWAYS_INLINE decltype(auto) operator()(const T& a) const noexcept {
+    return step_function(a);
+  }
+
+  template <typename T>
+  BLAZE_ALWAYS_INLINE decltype(auto) load(const T& a) const noexcept {
+    BLAZE_CONSTRAINT_MUST_BE_SIMD_PACK(T);
+    return step_function(a);
+  }
+};
+}  // namespace blaze
+
+template <typename VT, bool TF>
+// NOLINTNEXTLINE(readability-const-return-type)
+BLAZE_ALWAYS_INLINE decltype(auto) step_function(
```

Comment:
As far as I can tell, this function and the next are identical except for name, so why does this one need a NOLINT and the other does not?